internal/dataplane/kongstate/util.go

```diff
@@ -193,15 +199,16 @@ func kongPluginFromK8SPlugin(
 
 	return Plugin{
 		Plugin: plugin{
-			Name:   k8sPlugin.PluginName,
-			Config: config,
-
+			Name:         k8sPlugin.PluginName,
+			Config:       config,
 			RunOn:        k8sPlugin.RunOn,
 			Ordering:     k8sPlugin.Ordering,
 			InstanceName: k8sPlugin.InstanceName,
 			Disabled:     k8sPlugin.Disabled,
 			Protocols:    protocolsToStrings(k8sPlugin.Protocols),
 			Tags:         util.GenerateTagsForObject(&k8sPlugin),
+
+			KongVersion: kongVersion,
```

Comment:
```suggestion
```
How about keeping the version out of `KongState` struct (to keep it clean from feature flags/versions, etc., and leave it just for storing raw generated entities) and just passing the version to `Plugin.toKongPlugin` (propagating from `KongState.FillPlugins`? This way we could avoid e.g. filling this in `ConfigFetcher`.